rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,429 @@
+
+- [lib/attrsets](https://github.com/NixOS/nixpkgs/blob/master/lib/attrsets.nix)
+- [trivial-builders](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/trivial-builders.nix)
+- [stdenv/mkDerivation](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix)
+- [nixos/lib/make-disk-image](https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-disk-image.nix)
+
+# Design
+[design]: #detailed-design
+
+In the following we give a comprehensive overview to our decision that we've made. Detailed arguments for and against every decision can be found in the [Decisions](#Decisions) section
+
+## CommonMark
+
+**CommonMark is the content of all doc-comments.**
+
+Adopting CommonMark as the content for all doc-comments brings the benefits of widely accepted and understood documentation format in tech projects, while maintaining profitability and consistency within the Nix ecosystem by aligning with existing [NixOS/RFC-72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).
+
+## `/** */` is the doc-comment format
+
+The decision to use /** to start a doc-comment ensures a unique distinction from regular comments while still allowing seamless writing without IDE or editor support. This choice not only provides the best developer experience but also minimizes the need for additional tooling overhead.
+
+## Placement
+
+**The placment describes the relationship between doc-comments and the expression they are documenting.**
+
+- Doc-comments are placed before the documentable node. Only `WHITESPACES` are allowed in between. ([Examples](#basic-examples)) 
+    - `WHITESPACES` are: `[\n \r ' ' \t]`.
+
+- The documentation present before the `attribute path` describes the body of the attribute. ([Examples](#Attributes))
+    - In case placement is ambiguous, the one closer to the body has higher precedence. ([Examples](#ambiguous-placement))
+         
+- All partial functions of a curried lambda can share the same placement with the outermost lambda. ([Examples](#partial-lambda-functions))
+  
+> Note: Research of the RFC Sheperds Team showed that this allows for intuitive placements like are already done in nixpkgs.
+
+### Examples
+
+#### Basic examples
+
+Only whitespaces between the `documentable node` and the `doc-comment`
+
+```nix
+/**Doc for anonymous lambda function*/
+↓
+x: x;
+```
+
+```nix
+listToAttrs [
+  { name = "foo"; value = /**Documentation for '1'*/1; }
+]
+```
+
+#### Attributes 
+
+It is allowed to write the documentation before the attribute instead of placing it right before the body.
+
+```nix
+/**Doc for lambda function bound to a variable*/
+           ↓
+assigned = x: x;
+```
+
+```nix
+{
+    /**This documents the specialisation `map (x: x)` */
+          ↓
+    foo = map (x: x);          
+}
+```
+
+#### Ambiguous placement
+
+```nix
+/**Doc B*/
+int = /**Doc A*/1;
+
+# Documentation is 'Doc A' because it is directly next to the documentable body.
+```
```

Comment:
Was expecting an example with an attribute path syntax when you referred me here. Seems worth adding.

```
### Attribute path leaf
/** Doc */
a.b.c = 1;
```

`Doc` applies to `c` only, I suppose?

Reply:
Yep. As written only applies to the leaf. Maybe we should explicitly explain what it applies to.

Reply:
Oh Sorry. I think we did indeed forget to add this specific example. :+1: